testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/ResetPasswordTest.java

```diff
@@ -1084,4 +1086,140 @@ public void failResetPasswordServiceAccount() {
         loginPage.assertCurrent();
         assertEquals("Invalid username or password.", errorPage.getError());
     }
+
+    @Test
+    public void resetPasswordLinkNewTabAndProperRedirectAccount() throws IOException {
+        final String REQUIRED_URI = OAuthClient.AUTH_SERVER_ROOT + "/realms/test/account/applications";
+        final String REDIRECT_URI = getAccountRedirectUrl() + "?path=applications";
+        final String CLIENT_ID = "account";
+
+        try (BrowserTabUtil tabUtil = BrowserTabUtil.getInstanceAndSetEnv(driver)) {
```

Comment:
Is it possible to test the scenario with 2 browsers (driver and driver2 variables already available inside this test) instead of the different browser tabs as we discussed on the call? It seems to me the test with more browser tabs is quite error prone and there might be issues/instabilities for various browsers (which will occur especially in the environments like jenkins etc)

Reply:
As I mentioned above, simulate this case with two web drivers is quite difficult (at least for me). When I open a link from received email by _driver2_, there isn't the same environment like in new tab created by first _driver_ and this leads to a different behavior. The page isn't redirected back to the application, but there is only message 'Your account has been updated'. So, it's not the required behavior. 

At this time, I don't know, how to do that better. So, If you have better solution for that, you may share your hints. Thanks @mposolda

Reply:
Yes, true. Let's go with your approach then. It will be good to run some set of pipeline tests for various browsers to make sure that things are not broken for different browsers...